lib/utils/application_insights.ts

```diff
@@ -8,7 +8,17 @@ export interface ITelemetryParams {
   readonly serviceId?: ServiceId;
 }
 
+/**
+ * Get a new ApplicationInsights TelemetryClient.
+ *
+ * When using context and/or commonProperties
+ * the TelemetryClient shares them between all threads
+ * (ie. function handlers runs). That's why we must obtain
+ * a new instance every time we want to set common values
+ * which are valid only for one run.
+ */
 export function getApplicationInsightsTelemetryClient(
+  isProduction: boolean,
   params: ITelemetryParams,
```

Comment:
would be nice to have handlers take an `(ITelemetryParams, Record<string, string>) => ApplicationInsights.TelemetryClient` and have this method as `boolean => (ITelemetryParams, Record<string, string>) => ApplicationInsights.TelemetryClient`

Reply:
I don't get this, are you suggesting to curry the isProduction parameter ?

Reply:
yes